### PR TITLE
2253 redesign meterregistry

### PR DESCRIPTION
**Pull request by ericbottard**
This PR is on top of #2261, which is used as a way to prove that this PR does not introduce classloader issues.

Please focus on 127f69f when reviewing.

### CODE REVIEW

**Comment by ericbottard** [comment]
@simonbasle what do you think of this (assuming we're fine with #2261)

File: reactor-core/src/withMicrometerTest/java/reactor/core/scheduler/SchedulersMetricsTest.java [hunk 1]
```diff
@@ -32,31 +33,33 @@
...
-		simpleMeterRegistry = new SimpleMeterRegistry();
-		Metrics.addRegistry(simpleMeterRegistry);
+		registry = new SimpleMeterRegistry();
+		previousRegistry = reactor.util.Metrics.Configuration.useRegistry(registry);
 		Schedulers.enableMetrics();
 	}
 
 	@After
-	public void tearDown() {
+	public void teardDown() {
```
Comment (simonbasle) [inline reactor-core/src/withMicrometerTest/java/reactor/core/scheduler/SchedulersMetricsTest.java:51]: unnecessary rename / typo `tear`_`d`_`Down`

File: reactor-core/src/main/java/reactor/util/Metrics.java [hunk 2]
```diff
@@ -49,4 +52,23 @@ public static final boolean isInstrumentationAvailable() {
 		return isMicrometerAvailable;
 	}
 
+	public static class Configuration {
```
Comment (simonbasle) [inline reactor-core/src/main/java/reactor/util/Metrics.java:55]: Maybe rename `MicrometerConfiguration` ?

File: reactor-core/src/main/java/reactor/util/Metrics.java [hunk 3]
```diff
@@ -49,4 +52,23 @@ public static final boolean isInstrumentationAvailable() {
...
+
+		/**
+		 * The registry to use in reactor for metrics related purposes.
```
Comment (simonbasle) [inline reactor-core/src/main/java/reactor/util/Metrics.java:58]: Move on `getRegistry` and mention the default is `globalRegistry` ?

**Comment by simonbasle** [comment]
failing test is flakky. I attempted to fix with https://github.com/reactor/reactor-core/pull/2271

**Comment by simonbasle** [comment]
@ericbottard the PR I mentioned with the flakky test has been merged, can you rebase on top of latest `master` ? There are also a few small improvements to be done, see my review

File: reactor-core/src/main/java/reactor/core/publisher/Flux.java [hunk 4]
```diff
@@ -6072,6 +6073,11 @@ public int getPrefetch() {
 	 * Metrics are gathered on {@link Subscriber} events, and it is recommended to also
 	 * {@link #name(String) name} (and optionally {@link #tag(String, String) tag}) the
 	 * sequence.
+	 * <p>
+     * The {@link MeterRegistry} used by reactor can be configured via
```
Comment (ericbottard) [inline reactor-core/src/main/java/reactor/core/publisher/Flux.java:6077]: Fix

File: reactor-core/src/main/java/reactor/util/Metrics.java (hunk 2 above)
Comment (simonbasle) [reply reactor-core/src/main/java/reactor/util/Metrics.java:55]: (the rationale is that this strongly signals classloading of Micrometer. an added benefit is if we ever add support of other metrics frameworks)

**Pull request merged**